src/services/goals.js

```diff
@@ -403,23 +403,29 @@ export function reduceObjectivesForActivityReport(newObjectives, currentObjectiv
       // we can dedupe these using lodash
       exists.resources = uniqBy([
         ...exists.resources,
-        ...objective.activityReportObjectives[0].activityReportObjectiveResources.map(
-          (r) => r.dataValues,
-        ),
+        ...(objective.activityReportObjectives
```

Comment:
I'm realizing now that our fundamental problem is probably that everything is coming out of `getGoalsForReport`, which generates a giant fanout in the `ON "Goal"."id" = "objectives"."goalId"` join, because there are many goal IDs  joined to grants and recipients of the AR, all of which cross join to the objectives joined to topics, files and resources.

Reply:
Partially wrong - the fanout is in Topics

Reply:
Using the separate no longer is producing the fanout issue.